src/classes/issues-processor.ts

```diff
@@ -911,6 +932,59 @@ export class IssuesProcessor {
     return this.options.removeStaleWhenUpdated;
   }
 
+  private async _removeLabelsWhenUpdatedFromStale(
+    issue: Issue,
+    removeLabels: Readonly<string>[]
+  ): Promise<void> {
+    if (!removeLabels.length) {
+      return;
+    }
+
+    const issueLogger: IssueLogger = new IssueLogger(issue);
+
+    issueLogger.info(
+      `Removing all the labels specified via the ${this._logger.createOptionLink(
+        Option.RemoveLabelsWhenUpdatedFromStale
+      )}`
+    );
+
+    for (const label of removeLabels.values()) {
+      await this._removeLabel(issue, label);
+    }
+  }
+
+  private async _addLabelsWhenUpdatedFromStale(
+    issue: Issue,
+    addLabels: Readonly<string>[]
+  ): Promise<void> {
+    if (!addLabels.length) {
+      return;
+    }
+
+    const issueLogger: IssueLogger = new IssueLogger(issue);
+
+    issueLogger.info(
+      `Removing all the labels specified via the ${this._logger.createOptionLink(
+        Option.AddLabelsWhenUpdatedFromStale
+      )}`
+    );
+
+    try {
+      this._operations.consumeOperation();
+      this._statistics?.incrementAddedItemsLabel(issue);
+      await this.client.issues.addLabels({
```

Comment:
This specific code should not be executed in dry run mode (`this.options.debugOnly`)